moveit_demo_nodes/run_moveit_cpp/CMakeLists.txt

```diff
@@ -0,0 +1,39 @@
+cmake_minimum_required(VERSION 3.5)
+project(run_moveit_cpp)
+
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+# find dependencies
+find_package(ament_cmake REQUIRED)
+find_package(moveit_ros_planning_interface REQUIRED)
+# This shouldn't be necessary (required by moveit_simple_controller_manager)
+find_package(rosidl_default_runtime REQUIRED)
+find_package(Boost REQUIRED COMPONENTS system)
+# uncomment the following section in order to fill in
+# further dependencies manually.
+# find_package(<dependency> REQUIRED)
+
+add_executable(run_moveit_cpp src/run_moveit_cpp.cpp)
+ament_target_dependencies(run_moveit_cpp
+  moveit_ros_planning_interface
+)
```

Comment:
You should add `Boost` as target dependency

Reply:
I disagree since `Boost` is not required by `run_moveit_cpp`. Also, `Boost` is exported by `moveit_ros_planning_interface` which should provide all necessary targets. I ran into this issue on my system at first because `Boost` wasn't exported correctly. Having to depend on exported dependencies again seems like a bug in ament to me that I'm not able to reproduce on my machine.

Reply:
Ok, looks like this issue https://github.com/ament/ament_cmake/issues/199. Basically, components are not passed on to downstream dependencies atm...